sparql/src/main/java/org/apache/zeppelin/fuseki/SparqlInterpreter.java

```diff
@@ -0,0 +1,163 @@
+  public static final String SPARQL_REMOVE_DATATYPES = "sparql.removeDatatypes";
+
+  public SparqlInterpreter(Properties properties) {
+    super(properties);
+  }
+
+  @Override
+  public void open() {
+    LOGGER.info("Properties: {}", getProperties());
+  }
+
+  @Override
+  public void close() {
+  }
+
+  @Override
+  public InterpreterResult interpret(String queryString, InterpreterContext context) {
+    final String serviceEndpoint = getProperty(SPARQL_SERVICE_ENDPOINT);
+    final String replaceURIs = getProperty(SPARQL_REPLACE_URIS);
+    final String removeDatatypes = getProperty(SPARQL_REMOVE_DATATYPES);
+
+    LOGGER.info("SPARQL: Run Query '" + queryString + "' against " + serviceEndpoint);
+
+    if (StringUtils.isEmpty(queryString) || StringUtils.isEmpty(queryString.trim())) {
+      return new InterpreterResult(InterpreterResult.Code.SUCCESS);
+    }
+
+    Query query = null;
+    PrefixMapping prefixMapping = null;
+    try {
+      query = QueryFactory.create(queryString);
+      prefixMapping = query.getPrologue().getPrefixMapping();
+    } catch (QueryParseException e) {
+      LOGGER.error(e.toString());
+      return new InterpreterResult(
+        InterpreterResult.Code.ERROR,
+        "Error: " + e.getMessage());
+    }
+
+    try (QueryExecution qe =
+      QueryExecutionFactory.sparqlService(serviceEndpoint, query)) {
+      // execute query and get Results
+      ResultSet results = qe.execSelect();
+
+      // transform ResultSet to TSV-String
+      ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+      ResultSetFormatter.outputAsTSV(outputStream, results);
+      String tsv = new String(outputStream.toByteArray());
+
+      if (replaceURIs != null && replaceURIs.equals("true")) {
+        LOGGER.info("SPARQL: Replacing URIs");
+        tsv = replaceURIs(tsv, prefixMapping);
+      }
+
+      if (removeDatatypes != null && removeDatatypes.equals("true")) {
+        LOGGER.info("SPARQL: Removing datatypes");
+        tsv = removeDatatypes(tsv);
+      }
+
+      return new InterpreterResult(
+              InterpreterResult.Code.SUCCESS,
+              InterpreterResult.Type.TABLE,
+              tsv);
+    } catch (QueryExceptionHTTP e) {
+      LOGGER.error(e.toString());
+      int responseCode = e.getResponseCode();
+
+      if (responseCode == HttpStatus.SC_UNAUTHORIZED) {
+        return new InterpreterResult(
+          InterpreterResult.Code.ERROR,
+            "Unauthorized.");
+      } else if (responseCode == HttpStatus.SC_NOT_FOUND) {
+        return new InterpreterResult(
+          InterpreterResult.Code.ERROR,
+            "Endpoint not found, please check endpoint in the configuration.");
+      } else {
+        return new InterpreterResult(
+          InterpreterResult.Code.ERROR,
+            "Error: " + e.getMessage());
+      }
+    }
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) {
+  }
```

Comment:
`cancel` is unable to be supported ? Or do you have plan to support it ?

Reply:
done: added support for cancel